## 1. The World Bank's international debt data
<p>It's not that we humans only take debts to manage our necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives. <a href="https://www.worldbank.org">The World Bank</a> is the organization that provides debt to countries.</p>
<p>This notebook contains analysis on international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. 

<p> The analysis answers the following questions</p>
<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>
<p><img src="https://assets.datacamp.com/production/project_754/img/image.jpg" alt></p>

The internation_debt table contains the data on which I have performed this analysis. The first query reveals all the columns in the dataset.

In [2]:
%%sql
postgresql:///international_debt
    
SELECT 
    * 
FROM 
    international_debt
LIMIT
    10

10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.300000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.900000006')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.700000003'))]

## 2. Finding the number of distinct countries
<p>From the first ten rows, we can see the amount of debt owed by <em>Afghanistan</em> in the different debt indicators. But we do not know the number of different countries we have on the table. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator. </p>
<p>Without a count of unique countries, we will not be able to perform our statistical analyses holistically. </p>
<p>Next, we will extract the number of unique countries present in the table. </p>

In [4]:
%%sql
SELECT 
    COUNT(DISTINCT(country_name)) AS total_distinct_countries
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(124,)]

The query above shows that the total number of distinct countries in the dataset are <b>124</b>.
Next let's look at the list of these countries ordered alphabetically

In [15]:
%%sql
SELECT 
    DISTINCT(country_name) AS distinct_countries
FROM international_debt
ORDER BY distinct_countries;

 * postgresql:///international_debt
124 rows affected.


[('Afghanistan',),
 ('Albania',),
 ('Algeria',),
 ('Angola',),
 ('Armenia',),
 ('Azerbaijan',),
 ('Bangladesh',),
 ('Belarus',),
 ('Belize',),
 ('Benin',),
 ('Bhutan',),
 ('Bolivia',),
 ('Bosnia and Herzegovina',),
 ('Botswana',),
 ('Brazil',),
 ('Bulgaria',),
 ('Burkina Faso',),
 ('Burundi',),
 ('Cabo Verde',),
 ('Cambodia',),
 ('Cameroon',),
 ('Central African Republic',),
 ('Chad',),
 ('China',),
 ('Colombia',),
 ('Comoros',),
 ('Congo, Dem. Rep.',),
 ('Congo, Rep.',),
 ('Costa Rica',),
 ("Cote d'Ivoire",),
 ('Djibouti',),
 ('Dominica',),
 ('Dominican Republic',),
 ('Ecuador',),
 ('Egypt, Arab Rep.',),
 ('El Salvador',),
 ('Eritrea',),
 ('Eswatini',),
 ('Ethiopia',),
 ('Fiji',),
 ('Gabon',),
 ('Gambia, The',),
 ('Georgia',),
 ('Ghana',),
 ('Grenada',),
 ('Guatemala',),
 ('Guinea',),
 ('Guinea-Bissau',),
 ('Guyana',),
 ('Haiti',),
 ('Honduras',),
 ('IDA only',),
 ('India',),
 ('Indonesia',),
 ('Iran, Islamic Rep.',),
 ('Jamaica',),
 ('Jordan',),
 ('Kazakhstan',),
 ('Kenya',),
 ('Kosovo',),
 ('Kyrgyz Republic',),
 ('Lao PDR',),
 ('Least developed countries: UN classification',),
 ('Lebanon',),
 ('Lesotho',),
 ('Liberia',),
 ('Macedonia, FYR',),
 ('Madagascar',),
 ('Malawi',),
 ('Maldives',),
 ('Mali',),
 ('Mauritania',),
 ('Mauritius',),
 ('Mexico',),
 ('Moldova',),
 ('Mongolia',),
 ('Montenegro',),
 ('Morocco',),
 ('Mozambique',),
 ('Myanmar',),
 ('Nepal',),
 ('Nicaragua',),
 ('Niger',),
 ('Nigeria',),
 ('Pakistan',),
 ('Papua New Guinea',),
 ('Paraguay',),
 ('Peru',),
 ('Philippines',),
 ('Romania',),
 ('Russian Federation',),
 ('Rwanda',),
 ('Samoa',),
 ('Sao Tome and Principe',),
 ('Senegal',),
 ('Serbia',),
 ('Sierra Leone',),
 ('Solomon Islands',),
 ('Somalia',),
 ('South Africa',),
 ('South Asia',),
 ('Sri Lanka',),
 ('St. Lucia',),
 ('St. Vincent and the Grenadines',),
 ('Sudan',),
 ('Syrian Arab Republic',),
 ('Tajikistan',),
 ('Tanzania',),
 ('Thailand',),
 ('Timor-Leste',),
 ('Togo',),
 ('Tonga',),
 ('Tunisia',),
 ('Turkey',),
 ('Turkmenistan',),
 ('Uganda',),
 ('Ukraine',),
 ('Uzbekistan',),
 ('Vanuatu',),
 ('Venezuela, RB',),
 ('Vietnam',),
 ('Yemen, Rep.',),
 ('Zambia',),
 ('Zimbabwe',)]

## 3. Finding out the distinct debt indicators
<p> There is a column called <code>indicator_name</code> that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called <code>indicator_code</code> which symbolizes the category of these debts. Knowing about these various debt indicators will help us to understand the areas in which a country can possibly be indebted to. </p>

In [10]:
%%sql
SELECT
    DISTINCT(indicator_code) AS distinct_debt_indicators
FROM
    international_debt
ORDER BY
    distinct_debt_indicators;

 * postgresql:///international_debt
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

## 4. Totaling the amount of debt owed by the countries
<p>The financial debt of a particular country represents its economic state. Next, let's look at the global debt situation and find out the total amount of debt (in USD) that is owed by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.</p>

In [11]:
%%sql
SELECT 
    ROUND(SUM(debt)/1000000, 2) AS total_debt
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(Decimal('3079734.49'),)]

## 5. Country with the highest debt
<p>That is more than <em>3 million <strong>million</strong></em> USD, an amount which is really hard for us to fathom. </p>
<p>Next let's look at the country that owns the highest amount of debt along with the amount. 
<strong>Note</strong> that this debt is the sum of different debts owed by a country across several categories. This will help to understand more about the country in terms of its socio-economic scenarios.

In [16]:
%%sql
SELECT 
    country_name, 
    SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;

 * postgresql:///international_debt
1 rows affected.


[('China', Decimal('285793494734.200001568'))]

Next, let's look at the top 10 countries with most amount of debt

In [17]:
%%sql
SELECT 
    country_name, 
    SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 10;

 * postgresql:///international_debt
10 rows affected.


[('China', Decimal('285793494734.200001568')),
 ('Brazil', Decimal('280623966140.800007581')),
 ('South Asia', Decimal('247608723990.600003211')),
 ('Least developed countries: UN classification', Decimal('212880992791.900000988')),
 ('Russian Federation', Decimal('191289057259.200001943')),
 ('IDA only', Decimal('179048127207.299999298')),
 ('Turkey', Decimal('151125758035.300003616')),
 ('India', Decimal('133627060958.399997148')),
 ('Mexico', Decimal('124596786217.300001668')),
 ('Indonesia', Decimal('113435696693.499999149'))]

<p><em>China</em> comes out as the country with highest debt </p> 

## 6. Average amount of debt across indicators

<p> We already have an idea of the different debt indicators in which the countries owe their debts. Next, we find out on an average how much debt a country owes? This will give us a better sense of the distribution of the amount of debt across different indicators.</p>

In [19]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    AVG(debt) as average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

 * postgresql:///international_debt
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401.499193612')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194333.812658349')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041216.890243888')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983452.859836046')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694101.963265321')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024067.650806481')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290.398230108')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844.421518983')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963.083064523')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623947.653623188'))]

<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. <p>An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>

## 7. The highest amount of principal repayments

<p>We can investigate a bit more to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically. </p>

In [26]:
%%sql
SELECT 
    country_name, 
    indicator_name
FROM international_debt
WHERE debt = (SELECT 
                 MAX(debt)
             FROM international_debt
             WHERE indicator_code = 'DT.AMT.DLXF.CD');

 * postgresql:///international_debt
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)')]

<p>China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category. This is verified by <a href="https://data.worldbank.org/indicator/DT.AMT.DLXF.CD?end=2018&most_recent_value_desc=true">The World Bank</a>.

## 8. The most common debt indicator
<p>Long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out. </p>

In [4]:
%%sql
SELECT
    indicator_code,
    COUNT(indicator_code) AS indicator_count
FROM 
    international_debt
GROUP BY
    indicator_code
ORDER BY
    indicator_count DESC, indicator_code DESC
LIMIT
    20;


 * postgresql:///international_debt
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue.

## 9. Other viable debt issues and conclusion
<p>Next, let's find out the maximum amount of debt that each country has. With this, we will be in a position to identify the other plausible economic issues a country might be going through.</p>

In [5]:
%%sql
SELECT
    country_name,
    MAX(debt) AS maximum_debt
FROM 
    international_debt
GROUP BY
    country_name
ORDER BY
    maximum_debt DESC
LIMIT 10;


 * postgresql:///international_debt
10 rows affected.


[('China', Decimal('96218620835.699996948')),
 ('Brazil', Decimal('90041840304.100006104')),
 ('Russian Federation', Decimal('66589761833.5')),
 ('Turkey', Decimal('51555031005.800003052')),
 ('South Asia', Decimal('48756295898.199996948')),
 ('Least developed countries: UN classification', Decimal('40160766261.599998474')),
 ('IDA only', Decimal('34531188113.199996948')),
 ('India', Decimal('31923507000.799999237')),
 ('Indonesia', Decimal('30916112653.799999237')),
 ('Kazakhstan', Decimal('27482093686.400001526'))]

<p>In this notebook, we took a look at debt owed by countries across the globe. We extracted a few summary statistics from the data and unraveled some interesting facts and figures. We also validated our findings to make sure the investigations are correct.</p>

<p> Now we can answer the questions listed at the beginning of the notebook</p>
<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<p>More than <em>3 million <strong>million</strong></em> USD</p>
<li>Which country owes the maximum amount of debt and what does that amount look like?</li>
<p><strong>China</strong> owes the maximum amount of debt and the amount is <strong>285793494734.20</strong></p>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
<p>Refer to result in Section 6 of the notebook</p>
</ul>